### PR TITLE
feat(ui): add tooltips to sw360 enum values

### PR DESCRIPTION
**Pull request by maierthomas**
Show tooltip text for the sw360 enum values in datatables and tables.

The tooltips ca be configured in a CSS file
- sw360-portlet/webapp/css/tooltips.css

Closes sw360/sw360portal#394

### CODE REVIEW

**Comment by mcjaeger** [comment]
@adrian-evo if you would have time, mind a test?

**Comment by adrian-evo** [comment]
Noticed two issues so far:

1. Clearing state field is broken on edit release, it contains <span class=...
2. Moderation tab is broken

clearing

mod tab

**Comment by maierthomas** [comment]
1. changed to: bare output for the clearing state in the release edit mode.
2. fixed moderation tab

**Comment by mcjaeger** [comment]
@adrian-evo if you would have time, mind a re - test of the non working parts?

**Comment by adrian-evo** [comment]
The issues are fixed, however I don't see what is the change, i.e. there is no (i) icon that displays a tooltip. Can you show a sample screenshot about how it is supposed to work?

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/tags/DisplayEnumInfo.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import static com.google.common.base.Predicates.instanceOf;
+import static com.google.common.collect.Iterables.all;
+
+/**
+ * @author thomas.maier@evosoft.com
+ */
+public class DisplayEnumInfo extends SimpleTagSupport {
+
+    private static final String SW360_TOOLTIP = "sw360-tt";
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/tags/DisplayEnumInfo.java]: same constant as in DisplayEnum.class. Also, when you pull it out, it's probably better to rename it to more descriptive `SW360_TOOLTIP_CSS_CLASS` or something like that

File: frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf [hunk 2]
```diff
@@ -91,7 +91,7 @@
                 </core_rt:when>
                 <core_rt:otherwise>
                         '<div title="<sw360:out value='${vulnerabilityVerificationTooltips.get(vulnerability.externalId).get(vulnerability.intReleaseId)}'/>">'+
-                        '<sw360:DisplayEnum value="${vulnerabilityVerifications.get(vulnerability.externalId).get(vulnerability.intReleaseId)}"/>'+
+                        "<sw360:DisplayEnum value="${vulnerabilityVerifications.get(vulnerability.externalId).get(vulnerability.intReleaseId)}"/>"+
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf]: why did you change single quotes to double quotes in so many places? I don't see the reason.

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf [hunk 3]
```diff
@@ -33,6 +33,8 @@
                    value="<sw360:out value="${project.name}"/>"/>
             <img class="infopic" src="<%=request.getContextPath()%>/images/ic_info.png"
                  title="Name of the project"/>
+
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf]: unnecessary line breaks

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/tags/DisplayEnumInfo.java (hunk 1 above)
Comment (maierthomas) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/tags/DisplayEnumInfo.java]: Moved to Portal Constants and renamed it to TOOLTIP_CLASS__CSS

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf (hunk 3 above)
Comment (maierthomas) [reply frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf]: Removed unnecessary line breaks

File: frontend/sw360-portlet/src/main/webapp/css/tooltips.css [hunk 4]
```diff
@@ -0,0 +1,723 @@
...
+.sw360-tt-AttachmentType-LICENSE_AGREEMENT:hover:after {
+    content: "A ruling license agreement for this item, note that this could be for commercial software for example";
+}
+
+.sw360-tt-AttachmentType-SCREENSHOT:hover:after {
+    content: "Screenshot, usually screenshot of the Website with licensing information";
+}
+
+.sw360-tt-AttachmentType-README_OSS:hover:after {
+    content: "tbd";
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/css/tooltips.css]: tbd?

Comment (maierthomas) [reply frontend/sw360-portlet/src/main/webapp/css/tooltips.css]: removed text tbd and hide this tooltip

File: frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf (hunk 2 above)
Comment (maierthomas) [reply frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf]: To avoid problems if single quotes are used in the DisplayEnum function.
Earlier I generated a span element with single quotes in the class attribute.
As a result problems occurred and I changed the single quotes to double quotes.

Currently the DisplayEnum function uses double quotes for the class value.

Comment (alexbrdn) [reply frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf]: Why didn't you use single quotes for the class attribute in DisplayEnum? I mean, it's not that important, but that would contain the change to DisplayEnum, instead of spreading it to multiple files, would it not?

Comment (maxhbr) [reply frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf]: The correct solution would be to escape the result of `DisplayEnum`?

Comment (alexbrdn) [reply frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf]: Right

Comment (maierthomas) [reply frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf]: In my opinion it doesn't make to escape the DisplayEnum result here.
The enum values itself can not have single or double quotes in their names.
Also the tooltip text is defined in the CSS file and not in javascript.

In case of double quotes in the tooltip text you need to escape them in the CSS file.

Comment (maxhbr) [reply frontend/sw360-portlet/src/main/webapp/html/components/includes/components/vulnerabilities.jspf]: until someone toggles the qouting in `DisplayEnum` or maybe use both styles of qouting. Escaping is the safe way.

**Comment by mcjaeger** [comment]
@alexbrdn @maxhbr we discussed to create an issue for the escaping because it seems like for other tags the issue
